src/value-converters/highlight-text-value-converter.ts

```diff
@@ -30,6 +30,9 @@ export class HighlightTextValueConverter {
         }
 
         if (regExp) {
+            // value is sanitized, so the RegExp has to be sanitized as well to match the results
+            // this fixes the issue with &, <, >, "
+            regExp = new RegExp(this._htmlSanitizer.sanitize(regExp.source), regExp.flags);
```

Comment:
While working on the test, I came across something:
When passing a single unescaped RegExp control character, i.e. `(` **as a string, not as precompiled RegExp**, the value converter will try to create a RegExp from it and fail, throwing a syntax error. Is this expected behaviour?

When passing a string, we expect it to be some kind of search term to be highlighted and not an uncompiled RegExp, right?

I would suggest using escape-string-regexp or something to prevent something like this from happening.

Reply:
Yes. Sanitizing the regex is not the right choice here.

You should sanitize the RegEx before passing it to `highlight-text-value-converter`:
```typescript
import escapeStringRegexp from 'escape-string-regexp';

private _mySearchRegExp = new RegExp("(" + escapeStringRegexp(searchQuery) + ")", "ig");
```
```html
<span innerhtml="${property|highlightText:_mySearchRegExp}">&nbsp;</span>
```

Or when passing a string to `highlight-text-value-converter`.
```typescript
 else if (text && text.length > 0) {
    regExp = new RegExp("(" + escapeStringRegexp(text) + ")", "ig");
}
```

Reply:
Thanks for the response, took me a while to get to it again.

I implemented your last code example 1:1.
```
else if (text && text.length > 0) {
    regExp = new RegExp("(" + escapeStringRegexp(text) + ")", "ig");
}
```

The sanitization is still relevant. The issue it targets is unrelated, as demonstrated by the tests.

Reply:
Err... do I understand correctly that this is an open issue with this PR? (Kindly `Resolve conversation` resolved conversations.)
... Kind of itching to get this one merged - got time on hand today and saw the approval mark, but now I don't dare to.